### PR TITLE
Change the network lanes focus and exclude

### PR DESCRIPTION
**Pull request by SchSeba**
**What this PR does / why we need it**:
Using ginkgo flags I remove the skip functions.
Make focus and skip related to the lanes in the test.sh files.

So if Genie or Multus are not install the lane will fail.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Related to and issue of skipping multus tests because of daemonset name.

Example:

https://jenkins.ovirt.org/job/kubevirt_kubevirt_standard-check-pr/3347/artifact/ci_build_summary.html
check-patch.openshift-3.11-multus.el7.x86_64

```
S [SKIPPING] in Spec Setup (BeforeEach) [0.002 seconds]
Multus Networking
/root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:48
  VirtualMachineInstance with cni ptp plugin interface [BeforeEach]
  /root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:127
    should create a virtual machine with one interface
    /root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:137

    Skip multus tests that required multus cni plugin

    /root/go/src/kubevirt.io/kubevirt/tests/utils.go:2275
------------------------------
S [SKIPPING] in Spec Setup (BeforeEach) [0.002 seconds]
Multus Networking
/root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:48
  VirtualMachineInstance with cni ptp plugin interface [BeforeEach]
  /root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:127
    should create a virtual machine with one interface with network definition from different namespace
    /root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:154

    Skip multus tests that required multus cni plugin

    /root/go/src/kubevirt.io/kubevirt/tests/utils.go:2275
------------------------------
S [SKIPPING] in Spec Setup (BeforeEach) [0.002 seconds]
Multus Networking
/root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:48
  VirtualMachineInstance with cni ptp plugin interface [BeforeEach]
  /root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:127
    should create a virtual machine with two interfaces
    /root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:171

    Skip multus tests that required multus cni plugin

    /root/go/src/kubevirt.io/kubevirt/tests/utils.go:2275
------------------------------
S [SKIPPING] in Spec Setup (BeforeEach) [0.002 seconds]
Multus Networking
/root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:48
  VirtualMachineInstance with sriov plugin interface [BeforeEach]
  /root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:210
    should create a virtual machine with sriov interface
    /root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:218

    Skip multus tests that required multus cni plugin

    /root/go/src/kubevirt.io/kubevirt/tests/utils.go:2275
------------------------------
S [SKIPPING] in Spec Setup (BeforeEach) [0.002 seconds]
Multus Networking
/root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:48
  VirtualMachineInstance with sriov plugin interface [BeforeEach]
  /root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:210
    should create a virtual machine with two sriov interfaces referring the same resource
    /root/go/src/kubevirt.io/kubevirt/tests/vmi_multus_test.go:274

    Skip multus tests that required multus cni plugin

    /root/go/src/kubevirt.io/kubevirt/tests/utils.go:2275
```

**Release note**:

```release-note
Change network lanes focus and exclude 
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubevirt/kubevirt/2007)

### CODE REVIEW

File: tests/vmi_multus_test.go [hunk 1]
```diff
@@ -52,16 +52,13 @@ var _ = Describe("Multus", func() {
...
-	tests.PanicOnError(err)
-
 	nodeAffinity := &k8sv1.Affinity{
 		NodeAffinity: &k8sv1.NodeAffinity{
 			RequiredDuringSchedulingIgnoredDuringExecution: &k8sv1.NodeSelector{
 				NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
 					{
 						MatchExpressions: []k8sv1.NodeSelectorRequirement{
-							{Key: "kubernetes.io/hostname", Operator: k8sv1.NodeSelectorOpIn, Values: []string{nodes.Items[0].Name}},
+							{Key: "kubernetes.io/hostname", Operator: k8sv1.NodeSelectorOpIn, Values: []string{"node01"}},
```
Comment (cynepco3hahue) [inline tests/vmi_multus_test.go]: QE also runs our tests, so they will have different hostnames, and also ca we sure that we run on the schedulable node.

Comment (SchSeba) [reply tests/vmi_multus_test.go]: I just use the same configuration as the storage tests like 

```golang
func CreateHostPathPvWithSize(osName string, hostPath string, size string) {
	virtCli, err := kubecli.GetKubevirtClient()
	PanicOnError(err)

	quantity, err := resource.ParseQuantity(size)
	PanicOnError(err)

	hostPathType := k8sv1.HostPathDirectoryOrCreate

	name := fmt.Sprintf("%s-disk-for-tests", osName)
	pv := &k8sv1.PersistentVolume{
		ObjectMeta: metav1.ObjectMeta{
			Name: name,
			Labels: map[string]string{
				"kubevirt.io/test": osName,
			},
		},
		Spec: k8sv1.PersistentVolumeSpec{
			AccessModes: []k8sv1.PersistentVolumeAccessMode{k8sv1.ReadWriteOnce},
			Capacity: k8sv1.ResourceList{
				"storage": quantity,
			},
			PersistentVolumeReclaimPolicy: k8sv1.PersistentVolumeReclaimRetain,
			PersistentVolumeSource: k8sv1.PersistentVolumeSource{
				HostPath: &k8sv1.HostPathVolumeSource{
					Path: hostPath,
					Type: &hostPathType,
				},
			},
			StorageClassName: StorageClassHostPath,
			NodeAffinity: &k8sv1.VolumeNodeAffinity{
				Required: &k8sv1.NodeSelector{
					NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
						{
							MatchExpressions: []k8sv1.NodeSelectorRequirement{
								{
									Key:      "kubernetes.io/hostname",
									Operator: k8sv1.NodeSelectorOpIn,
									Values:   []string{"node01"},
								},
							},
						},
					},
				},
			},
		},
	}
```

Comment (SchSeba) [reply tests/vmi_multus_test.go]: maybe we need to introduce a function to find the right node and use it for all the affinity tests?

**Comment by SchSeba** [comment]
For reference 

https://jenkins.ovirt.org/job/kubevirt_kubevirt_standard-check-pr/3471//artifact/check-patch.k8s-multus-1.12.2.el7.x86_64/mock_logs/script/stdout_stderr.log

**Comment by SchSeba** [comment]
The lanes pick the right kubevirt-provider now remove the hold

/hold cancel

**Comment by SchSeba** [comment]
@cynepco3hahue, @slintes, @davidvossel  please take a look on this PR.

we have a major problem with our `test.sh` script the regex if didn't work and we always run on the default providers (k8s.1.11.0 and os-3.11.0)

look at the `TARGET` variable and then the `export KUBEVIRT_PROVIDER`

Example
```
TARGET=k8s-multus-1.12.2
+ export WORKSPACE=/home/jenkins/workspace/kubevirt_kubevirt_standard-check-pr/kubevirt
+ WORKSPACE=/home/jenkins/workspace/kubevirt_kubevirt_standard-check-pr/kubevirt
+ readonly ARTIFACTS_PATH=/home/jenkins/workspace/kubevirt_kubevirt_standard-check-pr/kubevirt/exported-artifacts
+ ARTIFACTS_PATH=/home/jenkins/workspace/kubevirt_kubevirt_standard-check-pr/kubevirt/exported-artifacts
+ readonly TEMPLATES_SERVER=https://templates.ovirt.org/kubevirt/
+ TEMPLATES_SERVER=https://templates.ovirt.org/kubevirt/
+ [[ k8s-multus-1.12.2 =~ openshift-.* ]]
+ [[ k8s-multus-1.12.2 =~ .*-1.10.4-.* ]]
+ [[ k8s-multus-1.12.2 =~ .*-multus-1.12.2-.* ]]
+ [[ k8s-multus-1.12.2 =~ .*-genie-1.11.1-.* ]]
+ export KUBEVIRT_PROVIDER=k8s-1.11.0
+ KUBEVIRT_PROVIDER=k8s-1.11.0
```

Another one:
```
TARGET=openshift-3.11-multus
+ export WORKSPACE=/home/jenkins/workspace/kubevirt_kubevirt_standard-check-pr/kubevirt
+ WORKSPACE=/home/jenkins/workspace/kubevirt_kubevirt_standard-check-pr/kubevirt
+ readonly ARTIFACTS_PATH=/home/jenkins/workspace/kubevirt_kubevirt_standard-check-pr/kubevirt/exported-artifacts
+ ARTIFACTS_PATH=/home/jenkins/workspace/kubevirt_kubevirt_standard-check-pr/kubevirt/exported-artifacts
+ readonly TEMPLATES_SERVER=https://templates.ovirt.org/kubevirt/
+ TEMPLATES_SERVER=https://templates.ovirt.org/kubevirt/
+ [[ openshift-3.11-multus =~ openshift-.* ]]
+ export KUBEVIRT_MEMORY_SIZE=6144M
+ KUBEVIRT_MEMORY_SIZE=6144M
+ [[ openshift-3.11-multus =~ .*-crio-.* ]]
+ [[ openshift-3.11-multus =~ .*-multus-.* ]]
+ export KUBEVIRT_PROVIDER=os-3.11.0
+ KUBEVIRT_PROVIDER=os-3.11.0
```

**Comment by SchSeba** [comment]
For reference

https://jenkins.ovirt.org/job/kubevirt_kubevirt_standard-check-pr/3506/artifact/ci_build_summary.html

File: automation/test.sh [hunk 2]
```diff
@@ -244,15 +244,18 @@ spec:
...
-elif [[ $TARGET =~ multus.* ]] || [[ $TARGET =~ genie.* ]]; then
-  # Run networking tests only (general networking, multus, genie, ...)
-  # If multus or genie is not present the test will  be skipped base on per-test checks
-  ginko_params="$ginko_params --ginkgo.focus=Networking|VMIlifecycle|Expose|Networkpolicy"
+elif [[ $TARGET =~ multus.* ]]; then
+  ginko_params="$ginko_params --ginkgo.focus=Multus|Networking|VMIlifecycle|Expose"
+elif [[ $TARGET =~ genie.* ]]; then
+  ginko_params="$ginko_params --ginkgo.focus=Genie|Networking|VMIlifecycle|Expose"
+else
+    ginko_params="$ginko_params --ginkgo.skip=Multus|Genie"
```
Comment (cynepco3hahue) [inline automation/test.sh]: double indent

File: automation/test.sh [hunk 3]
```diff
@@ -38,18 +38,18 @@ if [[ $TARGET =~ openshift-.* ]]; then
   # of old vms does not go fast enough they run out of memory.
   # To still allow continuing with the tests, give more memory in CI.
   export KUBEVIRT_MEMORY_SIZE=6144M
-  if [[ $TARGET =~ .*-crio-.* ]]; then
+  if [[ $TARGET =~ .*-crio.* ]]; then
```
Comment (cynepco3hahue) [inline automation/test.sh]: you can remove all `.*` from the end, but it does not critical

File: automation/test.sh (hunk 2 above)
Comment (SchSeba) [reply automation/test.sh]: Done

File: automation/test.sh (hunk 3 above)
Comment (SchSeba) [reply automation/test.sh]: Done

File: tests/vmi_multus_test.go [hunk 4]
```diff
@@ -45,15 +45,15 @@ const (
...
+var _ = Describe("Multus", func() {
 
 	flag.Parse()
 
 	virtClient, err := kubecli.GetKubevirtClient()
 	tests.PanicOnError(err)
 
-	nodes, err := virtClient.CoreV1().Nodes().List(v13.ListOptions{})
-	tests.PanicOnError(err)
+	nodes := tests.GetAllSchedulableNodes(virtClient)
```
Comment (cynepco3hahue) [inline tests/vmi_multus_test.go]: looks like you need to put it under `BeforeAll` method or remove
```
Expect(err).ToNot(HaveOccurred(), "Should list compute nodes")
```
from the `GetAllSchedulableNodes` method.

Comment (SchSeba) [reply tests/vmi_multus_test.go]: Thanks my bad. I update the PR now move this section into the `BeforeAll` I hope the tests will pass now

**Comment by SchSeba** [comment]
For reference 
https://jenkins.ovirt.org/job/kubevirt_kubevirt_standard-check-pr/3513/artifact/ci_build_summary.html

**Comment by SchSeba** [comment]
ci test please

**Comment by cynepco3hahue** [comment]
@SchSeba I can see a lot of failures under http://jenkins.ovirt.org/job/kubevirt_kubevirt_standard-check-pr/3515//artifact/check-patch.k8s-multus-1.12.2.el7.x86_64/mock_logs/script/stdout_stderr.log, can you please check it?

**Comment by cynepco3hahue** [comment]
ci test please

**Comment by SchSeba** [comment]
ci test please

**Pull request merged**